contracts/staking/test/utils/api_wrapper.ts

```diff
@@ -0,0 +1,254 @@
+        ): SignedStakingPoolApproval => {
+            const makerPrivateKey =
+                makerPrivateKeyIfExists !== undefined
+                    ? makerPrivateKeyIfExists
+                    : testUtilsConstants.TESTRPC_PRIVATE_KEYS[this._accounts.indexOf(makerAddress)];
+            const verifierAddress =
+                verifierAddressIfExists !== undefined ? verifierAddressIfExists : this.stakingProxyContract.address;
+            const chainId = chainIdIfExists !== undefined ? chainIdIfExists : constants.CHAIN_ID;
+            const approvalFactory = new ApprovalFactory(makerPrivateKey, verifierAddress, chainId);
+            const signedStakingPoolApproval = approvalFactory.newSignedApproval(poolId, makerAddress, signatureType);
+            return signedStakingPoolApproval;
+        },
+    };
+
+    private readonly _web3Wrapper: Web3Wrapper;
+    private readonly _accounts: string[];
+
+    constructor(
+        env: BlockchainTestsEnvironment,
+        accounts: string[],
+        ownerAddress: string,
+        stakingProxyContract: StakingProxyContract,
+        stakingContract: StakingContract,
+        zrxVaultContract: ZrxVaultContract,
+        ethVaultContract: EthVaultContract,
+        rewardVaultContract: StakingPoolRewardVaultContract,
+        zrxTokenContract: DummyERC20TokenContract,
+    ) {
+        this._web3Wrapper = env.web3Wrapper;
+        this._accounts = accounts;
+        this.stakingContract = stakingContract;
+        this.zrxVaultContract = zrxVaultContract;
+        this.ethVaultContract = ethVaultContract;
+        this.rewardVaultContract = rewardVaultContract;
+        this.zrxTokenContract = zrxTokenContract;
+
+        // disguise the staking proxy as a StakingContract
+        const decoderArtifacts = _.merge(artifacts, erc20Artifacts) as { [contractName: string]: ContractArtifact };
+        const logDecodeDependencies: { [contractName: string]: ContractAbi } = {};
+        if (Object.keys(decoderArtifacts) !== undefined) {
+            for (const key of Object.keys(decoderArtifacts)) {
+                logDecodeDependencies[key] = decoderArtifacts[key].compilerOutput.abi;
+            }
+        }
+        this.stakingProxyContract = new StakingContract(
+            stakingProxyContract.address,
+            env.provider,
+            {
+                ...txDefaults,
+                from: ownerAddress,
+                to: stakingProxyContract.address,
+                gas: 3000000,
+                gasPrice: 0,
+            },
+            logDecodeDependencies,
+        );
+    }
+}
+
+/**
+ * Deploys and configures all staking contracts and returns a StakingApiWrapper instance, which
+ * holds the deployed contracts and serves as the entry point for their public functions.
+ */
+export async function deployAndConfigureContractsAsync(
+    env: BlockchainTestsEnvironment,
+    ownerAddress: string,
+    erc20ProxyContract: ERC20ProxyContract,
```

Comment:
Should we just go ahead and deploy these ERC20/Proxy contracts here as well? Is there an integration test suite that doesn't just deploy standard implementations?

Reply:
good call